cdk/README.md

```diff
@@ -9,64 +9,147 @@ In this example we'll configure one Consul server in VPC with TLS and gossip enc
 * AWS CDK Toolkit, you can install it via: `npm install -g aws-cdk`
 
 ## Step 1: Create the project directory
-First create an empty directory on your system, initialize Typescript CDK project and install NPM packages.
+First create an empty directory on your system, initialize Typescript CDK project and install NPM packages. Replace $ACCOUNT and $REGION with your target AWS account id and region accordingly.
 
 ```
 mkdir app && cd app
 cdk init --language typescript
-cdk bootstrap aws://{ACCOUNT}/{REGION}
+cdk bootstrap aws://$ACCOUNT/$REGION
 
 npm install @aws-cdk/core @aws-cdk/aws-ec2 @aws-cdk/aws-ecs @aws-cdk/aws-iam @aws-cdk/aws-secretsmanager @aws-cdk-containers/ecs-service-extensions @aws-quickstart/ecs-consul-mesh-extension
 npm update
 ```
 
-## Step 2: Create and Deploy the VPC Environment 
-Next we're going to create the VPC Environment for launching the Consul server and the microservices.
-You will need to use your public IP and it's CIDR set to `$ALLOWED_IP_CIDR`.
-Example: `ALLOWED_IP_CIDR=$(curl -s ifconfig.me)/32`
+## Step 2: Define shared properties
+
+We are going to create and deploy the three separate stack for environment (VPC and ECS cluster), Consul server and microservices application stack. Create a new `lib/shared-props.ts` file with the following content. We are using this as shared properties between the stacks.
 
-### Create new environment
-Create a new `lib/shared-props.ts` file
 ```ts
 import * as cdk from '@aws-cdk/core';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as extensions from "@aws-cdk-containers/ecs-service-extensions";
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import { ConsulServer } from '../lib/consul-server'
 
 export interface EnvironmentInputProps extends cdk.StackProps {
   envName: string;
   allowedIpCidr: string;
 }
+
+export interface EnvironmentOutputProps extends cdk.StackProps {
+  envName: string;
+  vpc: ec2.Vpc;
+  serverSecurityGroup: ec2.SecurityGroup;
+  clientSecurityGroup: ec2.SecurityGroup;
+  ecsEnvironment: extensions.Environment;
+}
+
+export interface ServerInputProps extends cdk.StackProps {
+  envProps: EnvironmentOutputProps,
+  keyName: string,
+}
+
+export class ServerOutputProps {
+  serverTag: {[key: string]: string};
+  serverDataCenter: string;
+  agentCASecret: secretsmanager.ISecret;
+  gossipKeySecret: secretsmanager.ISecret;
+
+  constructor(serverScope: ConsulServer, agentCASecretArn: string, gossipKeySecretArn: string) {
+    this.serverTag = serverScope.serverTag;
+    this.serverDataCenter = serverScope.datacenter;
+    this.agentCASecret = secretsmanager.Secret.fromSecretAttributes(serverScope, 'ImportedConsulAgentCA', {
+      secretArn: agentCASecretArn
```

Comment:
`secretArn` is now deprecated so we can use `secretCompleteArn` instead.

Reply:
noted, I'll replace reference for `secretArn` with `secretCompleteArn`

Reply:
update : I changed it from using `SecretAttributes` to `ISecret` , thus this no longer applicable.